### PR TITLE
[feat] Add application server logic

### PR DESCRIPTION
**Pull request by nirajrao**
Splits the current directory structure into three parts:

1. `cmd` for the public-facing API.
2. `internal/pkg/http` for the application server.
3. `internal/cmd/api` for the internal server struct that is used by
   `cmd/api`.

The `server` struct exposes a visible method, `StartApplicationServer`
that can be used outside the `api` package. It runs a shutdown method in
a goroutine that waits to receive a `SIGINT` or `SIGTERM` signal. It
relies on `signal.Notify` to circumvent the *default* behavior for
synchronous signals in Go. Without this method call, the signal would be
converted into a runtime panic which would be caught by the main thread,
and the HTTP server would not be shutdown gracefully.

The `cmd/api/main.go` starts the application server and blocks on that
call. In the future, if we add an admin server, we should refactor this
functionality to start up everything relevant to running the server.

### CODE REVIEW

File: internal/cmd/api/server.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+
+	ihttp "card-match/internal/pkg/http"
```
[the remark this answers is not in the document]
Comment (abatilo) [reply internal/cmd/api/server.go:11]: I'd rename it. You're colliding with a super common standard library package `net/http`. The package itself might not be necessary but I'd also argue that it's the wrong abstraction to name after your package anyways. You're not dealing with the `http` protocol directly here.

File: internal/cmd/api/server.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+}
+
+func (s *server) StartApplicationServer() {
+	quit := make(chan os.Signal, 1)
+	signal.Notify(quit, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		<-quit
+		fmt.Print("Gracefully shutting down application server\n")
+		s.applicationServer.RegisterOnShutdown(func() {
```
Comment (abatilo) [inline internal/cmd/api/server.go:33]: Seems like the shutdown sequence is a strange place to register a shutdown callback

File: internal/cmd/api/server.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (nirajrao) [reply internal/cmd/api/server.go]: Decided to actually remove this package and keep everything in `internal/cmd/api`. I think I was using too much abstraction.

File: internal/cmd/api/server.go (hunk 2 above)
Comment (nirajrao) [reply internal/cmd/api/server.go]: Moved it to the application server handler.

**Pull request merged**